docs/book/en/src/dev/rust/wasinn.md

```diff
@@ -0,0 +1,158 @@
+* [`compute`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L165) the ML inference using the bound context
+* retrieve the inference result tensors using [`get_output`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L147-L160)
+
+You can find more detail about the WASI-NN proposal in [Reference](#Reference).
+
+In this section, we will use [an Rust example project](https://github.com/gusye1234/Wasi-NN-Examples) ![badge](https://github.com/gusye1234/Wasi-NN-Examples/actions/workflows/main.yaml/badge.svg) to demonstrate how to use the WASI-NN api and run a image classification demo.
+
+## Prerequisite
+
+Currently, WasmEdge used OpenVINO™ as the backend implementation. For this demo, you need to install [OpenVINO™](https://docs.openvino.ai/2021.4/openvino_docs_install_guides_installing_openvino_linux.html#)(2021) and [Rust](https://www.rust-lang.org/tools/install).
+
+By default, we don't enable any WASI-NN backend in WasmEdge. To enable the OpenVINO™ backend, we need to [building the WasmEdge from source](../../extend/build.md) with one more compiler flag `-DWASMEDGE_WASINN_BUILD_OPENVINO=ON` to enable the OpenVINO™ backend:
+
+```sh
+...
+cmake -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_WASINN_BUILD_OPENVINO=ON .. && make -j
+```
+(*Make sure you have configured the OpenVINO™ environment correctly with `source /opt/intel/openvino_2021/bin/setupvars.sh `* )
+
+For Rust, make sure to add `wasm32-wasi` target with
+
+```shell
+rustup target add wasm32-wasi
+```
+
+## Run
+
+Download the demo with:
+
+```sh
+git clone https://github.com/gusye1234/Wasi-NN-Examples.git
+```
+
+The demo will download the [fixtures](https://github.com/intel/openvino-rs/tree/main/crates/openvino/tests/fixtures/mobilenet) for the [Mobilenet](https://arxiv.org/abs/1704.04861), and run a image classification using the WASI-NN api:
+
+```sh
+cd Wasi-NN-Examples
+./build.sh <WASMEDGE RUNTIME PATH>
+# for example: 
+# ./build.sh "<path to WasmEdge repo>/build/tools/wasmedge/wasmedge"
+```
+
+If everything goes well, you should have the terminal output:
+
+```sh
+# massive rust compiling output above...
+Read graph XML, size in bytes: 143525
+Read graph weights, size in bytes: 13956476
+Loaded graph into wasi-nn with ID: 0
+Created wasi-nn execution context with ID: 0
+Read input tensor, size in bytes: 602112
+Executed graph inference
+   1.) [963](0.7113)pizza, pizza pie
+   2.) [762](0.0707)restaurant, eating house, eating place, eatery
+   3.) [909](0.0364)wok
+   4.) [926](0.0155)hot pot, hotpot
+   5.) [567](0.0153)frying pan, frypan, skillet
+```
+
+The generated `.wasm` file is under `./rust/examples/classification-example/build/wasi-nn-example.wasm`, we can see it imports the necessary WASI-NN functions using `wasm2wat`:
```

Comment:
Maybe you can give a link to `wasm2wat`.
You can say this: we can find that the WASM file imports the necessary WASI-NN functions by converting into WAT format with tools like `wasm2wat`: